src/Apps/Order/Components/BankAccountPicker.tsx

```diff
@@ -26,24 +30,21 @@ export const BankAccountPicker: FC<Props> = props => {
     order,
     bankAccountHasInsufficientFunds,
     setBankAccountHasInsufficientFunds,
-    onBankAccountContinue,
     setIsSavingPayment,
+    setBalanceCheckComplete,
+    setSelectedBankAccountId,
+    bankAccountSelection,
+    setBankAccountSelection,
   } = props
 
   const bankAccountsArray = extractNodes(bankAccounts)
-  const userHasExistingBankAccounts = bankAccountsArray.length > 0
-
-  type BankAccountSelectionType = "existing" | "new"
-
-  interface BankAccountSelection {
-    type: BankAccountSelectionType
-    id?: string
-  }
 
   const { submitMutation: setPaymentMutation } = useSetPayment()
 
   const handleContinue = async () => {
+    setBalanceCheckComplete(false)
```

Comment:
It's nice to see the consistency of the below sitting side by side

```
setBalanceCheckComplete(false)
setIsSavingPayment(true)
```

But there's a big difference between these two functions! One calls out _externally_ to the component and one is sets state _internally_. For an application _that lacks centralized state management_ to scale well over time crucial to maintain the distinctions between these two things at a glance. 

Why? Because one can be absolutely sure that the local `setIsSavingPayment` is being managed adequately here, where 
`setBalanceCheckComplete` -- being an external callback that's passed in -- is a wildcard. Was the right thing passed in at runtime to the prop? Is it defined? etc. Being a prop and external to the component, one can't control all of the details (even with TypeScript). So having idiomatic flags like `onBalanceCheckComplete` tells the future you that one needs to look at this piece of the code carefully, and in the future when for certain there will be a bug, a dev will know that this is external code executing, and we need to track it down.

Reply:
(This comment applies to all other locations in the PR where `onActionFoo` callbacks are seemingly being mixed in as local state)